### ISSUE TITLE
"explain_data_frame_analytics#Test non-empty data frame given body" doesn't match expected_memory_without_disk string

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#52576]
Stumbled into this on a PR CI run, reproduces reliably for me on master.  Will mute as well.

```
 ./gradlew ':x-pack:plugin:integTestRunner' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=ml/explain_data_frame_analytics/Test non-empty data frame given body}" -Dtests.seed=B99363F6D7C31F6F -Dtests.security.manager=true -Dtests.locale=ca -Dtests.timezone=America/Merida -Dcompiler.java=13 -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*
```
```
 2> java.lang.AssertionError: Failure at [ml/explain_data_frame_analytics:156]: memory_estimation.expected_memory_without_disk didn't match expected value:
    memory_estimation.expected_memory_without_disk: expected String [4kb] but was String [5kb]
        at __randomizedtesting.SeedInfo.seed([B99363F6D7C31F6F:31C75C2C793F7297]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:405)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:382)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1754)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:942)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:978)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:992)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:370)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:819)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:470)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:951)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:836)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:887)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:898)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
```

```
1> [2020-02-20T09:08:31,091][INFO ][o.e.x.t.r.XPackRestIT    ] [test] Stash dump on test failure [{
  1>   "stash" : {
  1>     "body" : {
  1>       "field_selection" : [
  1>         {
  1>           "name" : "x",
  1>           "mapping_types" : [
  1>             "float"
  1>           ],
  1>           "is_included" : true,
  1>           "is_required" : false,
  1>           "feature_type" : "numerical"
  1>         },
  1>         {
  1>           "name" : "y",
  1>           "mapping_types" : [
  1>             "float"
  1>           ],
  1>           "is_included" : true,
  1>           "is_required" : false,
  1>           "feature_type" : "numerical"
  1>         }
  1>       ],
  1>       "memory_estimation" : {
  1>         "expected_memory_without_disk" : "5kb",
  1>         "expected_memory_with_disk" : "5kb"
  1>       }
  1>     }
  1>   }
  1> }]
```

**Comment by droberts195** [comment]
@polyfractal this should have been fixed by #52452.  Are you sure you have an up-to-date master branch?

**Comment by polyfractal** [comment]
I think I was indeed slightly out of date.  After filing I noticed master was out of date slightly because pulling was broken on an outdated remote.  Pruned, re-running and it seems to pass.  Will close, sorry for the noise!
